src/Srclib/Types.hs

```diff
@@ -43,16 +44,27 @@ data SourceUnitDependency = SourceUnitDependency
   }
   deriving (Eq, Ord, Show)
 
-newtype AdditionalDepData = AdditionalDepData
-  {userDefinedDeps :: [SourceUserDefDep]}
+data AdditionalDepData = AdditionalDepData
+  { userDefinedDeps :: Maybe [SourceUserDefDep]
+  , remoteDeps :: Maybe [SourceRemoteDep]
```

Comment:
Is there a reason these are `Maybe`'s?  Is the empty list case different than the null case?  If not, these should just be lists.

Reply:
They are `Maybe`'s, because they are optional, and can be null.

> Is the empty list case different than the null case?

They would produce the same results in the Build, however if there are no `Remote` or `UserDefined` deps, I would rather have `AdditionalDeps` leave that field out completely, rather than having it point at an empty list. This makes more sense to me as we will only add fields that the user has provided explicitly, and additionally the type will more closely represent reality

Reply:
Good enough for me.